test/model.test.js

```diff
@@ -7062,6 +7057,63 @@ describe('Model', function() {
 
     });
 
+    it('saves documents with embedded discriminators (gh-15410)', async function() {
+      const requirementSchema = new Schema({
+        kind: { type: String, required: true },
+        quantity: Number,
+        notes: String
+      }, { _id: false, discriminatorKey: 'kind' });
+
+      const componentRequirementSchema = new Schema({
+        componentTest: 'ObjectId'
+      }, { _id: false });
+
+      const toolRequirementSchema = new Schema({
+        toolTest: 'ObjectId'
+      }, { _id: false });
+
+      const subRowSchema = new Schema({
+        requirements: [requirementSchema]
+      }, { _id: false });
+
+      const rowSchema = new Schema({
+        rows: [subRowSchema]
+      }, { _id: false });
+
+      const orderSchema = new Schema({
+        code: String,
+        rows: [rowSchema]
+      }, { timestamps: true });
+
+      const Requirement = requirementSchema;
+      Requirement.discriminators = {};
+      Requirement.discriminators['ComponentRequirement'] = componentRequirementSchema;
+      Requirement.discriminators['ToolRequirement'] = toolRequirementSchema;
```

Comment:
```suggestion
      requirementSchema.discriminators = {};
      requirementSchema.discriminators['ComponentRequirement'] = componentRequirementSchema;
      requirementSchema.discriminators['ToolRequirement'] = toolRequirementSchema;
```

Why re-assign if not necessary?